internal/handlers/user.go

```diff
@@ -39,63 +42,93 @@ func (api *API) UserIsCurrentUser(c *gin.Context) func(db *gorm.DB) *gorm.DB {
 	}
 }
 
-var noUUID = uuid.UUID{}
-
 func (api *API) createUserIfNotExists(ctx context.Context, id string, userName string) (uuid.UUID, error) {
 	ctx, span := tracer.Start(ctx, "createUserIfNotExists")
 	defer span.End()
 	span.SetAttributes(
 		attribute.String("user-id", id),
 		attribute.String("username", userName),
 	)
-	tx := api.db
 	var user models.User
-	res := tx.Unscoped().First(&user, "id = ?", id)
-	if res.Error == nil {
-
-		// Was the user was previously deleted... lets make him active again.
-		if user.DeletedAt.Valid {
-			user.DeletedAt = gorm.DeletedAt{}
-			res = tx.Unscoped().Model(&user).Update("DeletedAt", user.DeletedAt)
-			if res.Error != nil {
-				return noUUID, res.Error
+	var uuid uuid.UUID
+
+	err := api.db.WithContext(ctx).Transaction(func(tx *gorm.DB) error {
```

Comment:
use `api.transaction` instead of `api.db.WithContext(ctx).Transaction` it does the same thing an works better with cockroachdb in case we ever decide to support it.

    err := api.transaction(ctx, func(tx *gorm.DB) error {